### PR TITLE
Ft user can edit post 163515970

### PR DESCRIPTION
**Pull request by niomwungeri-fabrice**
#### What does this PR do?
Combined 4 endPoints:
`PUT api/v1/posts/:postId/content` Edit a post
`GET /api/v1/posts/:postId` Fetch a specific post
`GET /api/v1/posts` Fetch all posts
`GET /api/v1/user/posts` Fetch spefic posts for a specific user

#### How should this be manually tested?
* Clone the repository and run the dependencies installation command `yarn` or `npm install`
* Run `cp sample.env ./.env` to create the `.env` file and modify it to meet your configs
* Run the command to start the server `yarn server` or `npm run server`
* Open your favorite `REST CLIENT TOOL` (usually `postman`)
* Try requesting from the following Above APIs:
#### What are the relevant pivotal tracker stories?
#163515970

### CODE REVIEW

File: controllers/posts.js [hunk 1]
```diff
@@ -30,7 +33,80 @@ export default class PostController {
...
+        post: postResponse,
+      });
+    } catch (error) {
+      return res.status(500).send({ message: error, status: 500 });
+    }
+  }
+
+  static async getAllPost(req, res) {
+    try {
+      const postResponse = await Post.findAll();
```
Comment (abayo-luc) [inline controllers/posts.js]: you can include author details on each post.

File: controllers/users.js [hunk 2]
```diff
@@ -98,9 +98,28 @@ export default class UserController {
...
+    const userId = req.user.id;
+    Post.findAll({ where: { userId } })
+      .then((dataValues) => {
+        if (dataValues.length === 0) {
+          return res
+            .status(404)
+            .send({ message: 'The post does not exist', status: 404 });
+        }
+        return res.status(200).send({
+          message: 'A post fetched successfully',
```
Comment (abayo-luc) [inline controllers/users.js]: Posts fetched successfully

File: controllers/users.js [hunk 3]
```diff
@@ -98,9 +98,28 @@ export default class UserController {
...
   }
 
+  static getMyPosts(req, res) {
+    const userId = req.user.id;
+    Post.findAll({ where: { userId } })
+      .then((dataValues) => {
+        if (dataValues.length === 0) {
+          return res
+            .status(404)
+            .send({ message: 'The post does not exist', status: 404 });
```
Comment (abayo-luc) [inline controllers/users.js]: message:"User doesn't have any post"

File: controllers/posts.js (hunk 1 above)
Comment (niomwungeri-fabrice) [reply controllers/posts.js]: @abayo-luc  Done!

File: controllers/users.js (hunk 2 above)
Comment (niomwungeri-fabrice) [reply controllers/users.js]: @abayo-luc Done!

File: controllers/users.js (hunk 3 above)
Comment (niomwungeri-fabrice) [reply controllers/users.js]: @abayo-luc Done

**Pull request merged**